### PR TITLE
Catch2 contents removed from external\catch2 path

### PR DESCRIPTION
**Pull request by agarwalishita**
## Description

Catch2 was getting synced at 2 paths:
external\catch2
external\usersim\external\catch2

So, the former submodule has been deleted as part of this PR
and the source path used is : external\usersim\external\catch2 for creating build files

### CODE REVIEW

**Comment by agarwalishita** [comment]
@microsoft-github-policy-service agree

**Comment by dthaler** [comment]
You just need to remove catch2 from the .gitmodules file in the root of this repository, and change any include paths based on any compilation errors that occur to point to the other path

File: scripts/initialize_ebpf_repo.ps1 [hunk 1]
```diff
@@ -18,7 +18,8 @@ Set-Location $scriptPath\..
 # Define the commands to run
 $cmakeCommonArgs = "-G 'Visual Studio 17 2022' -A $Architecture"
 $commands = @(
-    "git submodule update --init --recursive",
+    # Update all submodules except the Catch2 within usersim
+    "git submodule update --init --recursive -- ':(exclude)external/usersim/external/Catch2'",
```
Comment (dthaler) [inline scripts/initialize_ebpf_repo.ps1:22]: Shouldn't need this

**Comment by agarwalishita** [comment]
> You just need to remove catch2 from the .gitmodules file in the root of this repository, and change any include paths based on any compilation errors that occur to point to the other path

In that case, if usersim stops depending on catch2, won't this repo start breaking?

**Comment by dthaler** [comment]
> > You just need to remove catch2 from the .gitmodules file in the root of this repository, and change any include paths based on any compilation errors that occur to point to the other path
> 
> In that case, if usersim stops depending on catch2, won't this repo start breaking?

Usersim should not stop depending on catch2, it should not be removed from usersim.
ebpf-for-windows should use the one exposed by usersim so this repo won't break.

**Comment by agarwalishita** [comment]
> > > You just need to remove catch2 from the .gitmodules file in the root of this repository, and change any include paths based on any compilation errors that occur to point to the other path
> > 
> > 
> > In that case, if usersim stops depending on catch2, won't this repo start breaking?
> 
> Usersim should not stop depending on catch2, it should not be removed from usersim. ebpf-for-windows should use the one exposed by usersim so this repo won't break.

Sure, thanks for confirming, will make the changes accordingly

**Comment by agarwalishita** [comment]
Created a new PR https://github.com/microsoft/ebpf-for-windows/pull/3973, hence closing this PR

**Pull request closed without merging**